lib/shopify-cli/commands.rb

```diff
@@ -8,23 +8,23 @@ module Commands
     )
     @core_commands = []
 
-    def self.register(const, cmd, path = nil)
+    def self.register(const, cmd, path = nil, is_core = false)
       autoload(const, path) if path
```

Comment:
This shouldn't work. Or rather it is adding an extra `autoload` to constants that will never load.

Reply:
This autoload line is basically exactly why I didn't use the register method

Reply:
The autoload line is only called if path is not nil.  In `ProjectType.register_command`, no path is passed in, so no autoload is called.  The pattern we've established for project types (in the `cli.rb` files) is:
```
module ProjectTypeName
  class Project < ShopifyCli::ProjectType
    creator 'name', 'ProjectTypeName::Commands::Create'

    register_command('ProjectTypeName::Commands::CommandA', "command_a")
    register_command('ProjectTypeName::Commands::CommandA', "command_b")
    :
  end

  # define/autoload project specific Commads
  module Commands
    autoload :Create,   Project.project_filepath('commands/create')
    autoload :CommandA, Project.project_filepath('commands/command_a')
    autoload :CommandB, Project.project_filepath('commands/command_b')
    :
  end

  # define/autoload project specific Tasks
  module Tasks
  end

  # define/autoload project specific Forms
  module Forms
  end
end
```
The only place where we pass a `path` to the `register()` method is inside `ShopifyCli::Commands` itself (i.e., when registering the core commands)